src/js/va-letters/containers/VALettersApp.jsx

```diff
@@ -1,19 +1,60 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import { connect } from 'react-redux';
 
+import RequiredLoginView from '../../common/components/RequiredLoginView';
+
+// This needs to be a React component for RequiredLoginView to pass down
+// the isDataAvailable prop, which is only passed on failure.
+function AppContent({ children, isDataAvailable }) {
+  const unregistered = isDataAvailable === false;
+  let view;
+
+  if (unregistered) {
+    view = (
+      <h4>
+        To view and download your VA letters, you need something something something.
+      </h4>
+    );
+  } else {
+    view = children;
+  }
+
+  return (
+    // TODO: add HTML and CSS when design has settled
+    <div className="row">
+      {view}
+    </div>
+  );
+}
+
 class VALettersApp extends React.Component {
   render() {
-    const { children } = this.props;
     return (
-      <div>{children}</div>
+      <RequiredLoginView
+          authRequired={3}
+          serviceRequired={"evss-letters"}
+          userProfile={this.props.profile}
+          loginUrl={this.props.loginUrl}
+          verifyUrl={this.props.verifyUrl}>
+        <AppContent>
+          {this.props.children}
```

Comment:
I don't totally understand how this works. Given the above definition of `<AppContent>`, I would have thought `children` would have to be passed as a prop, not as nested child content, if that makes sense. So it would have looked like this:
```
<App Content children={this.props.children} />
```
Googling for information on this isn't yielding anything fruitful so far. Do you happen to know how this works?